libtransact/src/state/merkle.rs

```diff
@@ -209,8 +209,11 @@ impl MerkleDatabase {
             // deleting a parent
             let successor = change_log.successors.pop().unwrap();
 
+            let mut deletions = successor.deletions.clone();
```

Comment:
Instead of cloning this, just make the successor `mut`, and push the parent on the original collection.